sdk/appconfig/azappconfig/README.md

```diff
@@ -0,0 +1,298 @@
+
+#### Create a client
+Constructing the client requires your vault's connection string, which you can get from the Azure Portal.
+```Bash
+export APPCONFIGURATION_CONNECTION_STRING="Endpoint=https://my-app-config.azconfig.io;Id=...;Secret=..."
+```
+
+```go
+import (
+    "os"
+    "github.com/Azure/azure-sdk-for-go/sdk/appconfig/azappconfig"
+)
+
+client, err = azappconfig.NewClientFromConnectionString(os.Getenv("APPCONFIGURATION_CONNECTION_STRING"), nil)
+```
+
+Or, using Default Azure Credential from Azure Identity:
+
+```go
+import (
+    "github.com/Azure/azure-sdk-for-go/sdk/appconfig/azappconfig"
+    "github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+)
+
+credential, err := azidentity.NewDefaultAzureCredential(nil)
+
+client, err = azappconfig.NewClient("https://my-app-config.azconfig.io", credential, nil)
+```
+
+## Examples
+This section contains code snippets covering common tasks:
+* [Add a configuration setting](#add-a-configuration-setting "Add a configuration setting")
+* [Get a configuration setting](#get-a-configuration-setting "Get a configuration setting")
+* [Set a configuration setting](#set-a-configuration-setting "Set a configuration setting")
+* [Set a configuration setting read only](#set-a-configuration-setting-read-only "Set a configuration setting read only")
+* [List configuration setting revisions](#list-configuration-setting-revisions "List configuration setting revisions")
+* [Delete a configuration setting](#set-a-configuration-setting "Delete a configuration setting")
+
+### Add a configuration setting
+
+```go
+import (
+    "fmt"
+    "os"
+
+    "github.com/Azure/azure-sdk-for-go/sdk/appconfig/azappconfig"
+    "github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+)
+
+func ExampleAddConfigurationSetting() {
+    connectionString := os.Getenv("APPCONFIGURATION_CONNECTION_STRING")
+    client, err = azappconfig.NewClientFromConnectionString(connectionString, nil)
+    if err != nil {
+        panic(err)
+    }
+
+    // Create configuration setting
+    resp, err := client.AddSetting(context.TODO(), azappconfig.Setting{Key: to.StringPtr("key"), Label: to.StringPtr("label"), Value: to.StringPtr("value")}, nil)
+    if err != nil {
+        panic(err)
+    }
+    fmt.Println(*resp.Key)
+    fmt.Println(*resp.Label)
+    fmt.Println(*resp.Value)
+}
+```
+
+### Get a configuration setting
+
+```go
+import (
+    "fmt"
+    "os"
+
+    "github.com/Azure/azure-sdk-for-go/sdk/appconfig/azappconfig"
+    "github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+)
+
+func ExampleGetConfigurationSetting() {
+    connectionString := os.Getenv("APPCONFIGURATION_CONNECTION_STRING")
+    client, err = azappconfig.NewClientFromConnectionString(connectionString, nil)
+    if err != nil {
+        panic(err)
+    }
+
+    // Get configuration setting
+    resp, err := client.GetSetting(context.TODO(), azappconfig.Setting{Key: to.StringPtr("key"), Label: to.StringPtr("label")}, nil)
```

Comment:
Does 'key' need to be defined as a string pointer? I wouldn't think it'd be optional at all (label is, however).

Reply:
Yes - because `Setting` is a return type, and you may ask for the list of items via `ListRevisions`, and ask it to not include `Key` column. In that case, the service will return `Setting` with `nil` `Key`.

Reply:
But this argument is for getting a setting - it has different semantics than ListSetting().

Is there a reason we're not just using 'key' as a positional argument?

Reply:
Then the key will be duplicated in the `Setting`, and as a parameter, I think it will be confusing.
If you look at the Java or .NET SDK, they have either methods with this semantics, taking `Setting`, or an overload with string key and optional string label, but not a mix.

Reply:
(this isn't a "delay this preview" thing. But let's discuss it for the next one)

Reply:
(if you don't mind filing an issue)

Reply:
As of ideas, initially, I had `Setting` being a struct with private fields, which you could only create as `NewSetting()` with string key parameter. It had getters to get fields as pointers, and "Setters", such as `WithLabel()`, `WithValue()` also taking strings and not pointers. Not all fields were writeable - the same as in other SDKs. And it was not possible then for user to create a setting without key. `With...()` were retutning a copy of the struct, with the corresponding field being set. But I was told such API is not Go-like, where everything is much simpler, uses bare structs, pointers, relies on comments etc.

The issue is: #17264